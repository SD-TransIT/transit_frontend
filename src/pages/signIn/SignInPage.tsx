import React from 'react';

import { useCookies } from 'react-cookie';
import Select from 'react-select';

import SignInForm from 'components/forms/signInForm';

export default function SignInPage() {
  const [langCookies, setLangCookie] = useCookies(['language']);

  const options = [
    { value: 'en', label: 'English' },
    { value: 'fr', label: 'French' },
    { value: 'pt', label: 'Portugese' },
  ];

  const onLocaleChange = (event: any) => {
    setLangCookie('language', { value: event.value, label: event.label }, { path: '/', sameSite: true });
  };

  return (
    <div className="flex h-screen w-screen bg-transit-grey" data-testid="sign-in-page">
      <div className="sm:w-1/2 bg-hero-image bg-no-repeat bg-cover" />
      <div className="w-full sm:w-1/2">
        <div className="flex flex-col h-screen">
          <div className="flex justify-end w-full pr-10 pt-10">
            <Select<Object>
              defaultValue={langCookies.language?.label}
              onChange={onLocaleChange}
              options={options}
              className="shadow-dateInput rounded w-1/5"
              placeholder={langCookies.language?.label}
            />
          </div>
          <SignInForm />
        </div>
      </div>
    </div>
  );
}
