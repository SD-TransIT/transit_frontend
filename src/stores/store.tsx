import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';

import rootReducer from 'stores/reducers/rootReducer';
import rootSaga from 'stores/sagas/rootSaga';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: rootReducer,
  middleware: [sagaMiddleware],
});

sagaMiddleware.run(rootSaga);

export default store;
