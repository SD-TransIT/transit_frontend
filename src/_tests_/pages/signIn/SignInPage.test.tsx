import React from 'react';

import { render, screen } from '@testing-library/react';
import SignInPage from 'pages/signIn/SignInPage';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from 'store';

describe('Unit tests for the SignInPage component.', () => {
  const SIGN_IN_PAGE_TEST_ID = 'sign-in-page';

  test('Should match snapshot.', () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[{ pathname: '/sign_in', search: '/' }]}>
          <SignInPage />
        </MemoryRouter>
      </Provider>,
    );

    const signInPage = screen.getByTestId(SIGN_IN_PAGE_TEST_ID);
    expect(signInPage).toBeInTheDocument();
    expect(signInPage).toMatchSnapshot();
  });
});
