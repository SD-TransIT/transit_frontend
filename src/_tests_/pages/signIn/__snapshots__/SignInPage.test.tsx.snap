// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the SignInPage component. Should match snapshot. 1`] = `
<div
  class="flex h-screen w-screen bg-transit-grey"
  data-testid="sign-in-page"
>
  <div
    class="sm:w-1/2 bg-hero-image bg-no-repeat bg-cover bg-center"
  />
  <div
    class="w-full sm:w-1/2"
  >
    <div
      class="form-floating"
    >
      <label
        for="floatingInput"
      >
        <input
          class="form-control"
          id="floatingInput"
          name="username"
          placeholder="Username"
        />
        Username
      </label>
    </div>
    <div
      class="form-floating mt-3"
    >
      <label
        for="floatingPassword"
      >
        <input
          class="form-control"
          id="floatingPassword"
          name="password"
          placeholder="Password"
          type="password"
        />
        Password
      </label>
    </div>
    <button
      class="w-100 btn btn-lg btn-warning"
      type="button"
    >
      Sign in
    </button>
  </div>
</div>
`;
